File: Qytetet_Ruby/pkg/Qytetet_Personal-0.0.1/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Dado
  
  require "singleton"
  include Singleton
  
  attr_accessor:valor
  
  def initialize
    @valor=0
  end
  
  def tirar
    @valor = rand(5)+1
    return @valor
  end
  
  def get_valor
    return @valor
  end
end

File: src/modeloqytetet/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.Random;
/**
 *
 * @author joseng2709
 */
public class Dado {
    private static Dado dado;
    
    int valor;
    
    public Dado(){
        dado = null;
        valor=0;
    }
    
    public static Dado getInstance(){
       if(dado == null){
           dado = new Dado();
    }
    return dado;
}
    
    public int Tirar(){
        Random rand = new Random();
        valor = rand.nextInt(6)+1;
        
        return valor;
    }
    public int getValor(){
        return valor;
    }
}
